public/app/features/alerting/unified/components/import-to-gma/NamespaceAndGroupFilter.tsx

```diff
@@ -35,14 +32,21 @@ export const NamespaceAndGroupFilter = ({ rulesSourceName }: Props) => {
     [namespaceGroups]
   );
 
-  const groupOptions: Array<SelectableValue<string>> = useMemo(
+  const groupOptions: Array<ComboboxOption<string>> = useMemo(
     () => (namespace && namespaceGroups.get(namespace)?.map((group) => ({ label: group, value: group }))) || [],
     [namespace, namespaceGroups]
   );
 
+  useEffect(() => {
+    // Reset namespace/group if datasource changes
+    setValue('namespace', '');
+    setValue('ruleGroup', '');
+  }, [rulesSourceName, setValue]);
```

Comment:
e.g., select datasource, then namespace/group, then change your mind about datasource - we need to reset the dropdowns

Reply:
💯